### PR TITLE
Add zoom to timeseries

### LINKED ISSUE #1168: Experiment Tracking - Time Series Plots - Build zoom functionalities
## Context
This feature request is an output from the experiment tracking User Testing Synthesis [#1627](https://github.com/kedro-org/kedro/issues/1627#top). User wishes to be able to perform 3 different zoom functionalities for further investigations in plots. 

This is important for focusing a part of the plot across the screen, and using the shift scroll functionality to quickly move to the end of the plot.

## Supporting quotes
“Ability to zoom in and out in the graphs to show more of a specific period." 

“If this goes on and on to the right or to the left, I would expect to be able to shift scroll to go to the right, and to the left. So basically the three scroll functionalities, the normal scroll, the control, and the shift.”

### PR DESCRIPTION
**Pull request by Huongg**
Signed-off-by: Cvetanka <cventanka_nechevska@external.mckinsey.com>

## Description

Resolves #1168 

## Development notes

Implement brushing for zooming, on doubleClick zoom is reset to the initial scale:

https://user-images.githubusercontent.com/88193161/206024061-d07fc6ec-3bfa-4234-bf60-a9d259c9c217.mov

What should be improved/fixed:

- ~onZoom path line is out of the chart on the left side:~
- ~brush is not showing on each chart~
- ~reset zoom when new run is selected~

- I've changed the color of the selected area: `$ocean-600: #052f41;`

Screenshot 2022-12-07 at 14 54 38

Screenshot 2022-12-07 at 14 55 45

## Checklist

- [x] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [ ] Added tests to cover my changes

### CODE REVIEW

File: src/components/experiment-tracking/time-series/time-series.js
Comment (tynandebold) [inline src/components/experiment-tracking/time-series/time-series.js:81]: Is this still needed? Can remove it if it's not.

File: src/components/experiment-tracking/time-series/time-series.js
Comment (tynandebold) [inline src/components/experiment-tracking/time-series/time-series.js:156]: Same as above: is this still needed? Can remove it if it's not.

File: src/components/experiment-tracking/time-series/time-series.scss
Comment (tynandebold) [inline src/components/experiment-tracking/time-series/time-series.scss:122]: What's this doing exactly?

**Review by tynandebold — CHANGES_REQUESTED** [review]
I'm liking where this is going. The ability to zoom in adds a lot of value, and I think we should make it a first-class feature. We'll need to finesse it some, as you've already pointed out in your PR comments:

- The zoom functionality should be available on **all** charts, not just the first. I'm guessing you'll have to attach it to each chart instance, because right now it's only on the first.
- We need to find a way to make sure the trend line (the horizontal line that connects all the runs) doesn't extend beyond the yAxis.
- When zoomed in and when a user is hovering over runs list cards, they could easily be hovering over something that's no longer displayed on the chart because of how zoomed in we are. To avoid this in one way, I think we should reset the zoom each time a user selects a new run from the runs list
- When clicking and dragging to select a brush area to zoom into, it's hard to see the vertical run lines. That probably doesn't need a background color.

Let me know what you think. If you fix the points you've already raised and that I've added to here it should be just about ready to go.

File: src/components/experiment-tracking/time-series/time-series.js
Comment (cvetankanechevska) [reply src/components/experiment-tracking/time-series/time-series.js:81]: Yes, this will be removed.

File: src/components/experiment-tracking/time-series/time-series.scss
Comment (cvetankanechevska) [reply src/components/experiment-tracking/time-series/time-series.scss]: This is why I changed the default style for the brush, the bottomLine is under the xAxis:

Screenshot 2022-12-07 at 13 31 17

**Review by rashidakanchwala — APPROVED** [review]
Loving it. Great work .. thanks @cvetankanechevska !

File: src/components/experiment-tracking/time-series/time-series.js
Comment (tynandebold) [inline src/components/experiment-tracking/time-series/time-series.js:10]: Nit: can you move this up to where the other JS imports are?

File: src/components/experiment-tracking/time-series/time-series.js
Comment (tynandebold) [inline src/components/experiment-tracking/time-series/time-series.js:19]: Nit: please move this up to where the other `useState` hooks are. Also, this should probably have a default value.

File: src/components/experiment-tracking/time-series/time-series.js
Comment (tynandebold) [inline src/components/experiment-tracking/time-series/time-series.js:130]: What are we setting the value to here? `false`?

File: src/components/experiment-tracking/time-series/time-series.js
Comment (tynandebold) [inline src/components/experiment-tracking/time-series/time-series.js:204]: Same here: what are we setting the value to? `false`?

**Review by tynandebold — APPROVED** [review]
It's working beautifully.

A couple small nits and questions. Approving now though! Please update with my comments before merging.

File: src/components/experiment-tracking/time-series/time-series.js
Comment (tynandebold) [reply src/components/experiment-tracking/time-series/time-series.js:130]: Can probably set the state to `undefined` then.

**Pull request merged**
